packages/orbit-components/src/InputField/index.tsx

```diff
@@ -182,41 +181,47 @@ const InputField = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
             ? "text-form-element-disabled-foreground"
             : "text-form-element-filled-foreground",
         )}
-        ref={label ? null : labelRef}
       >
-        {inlineLabel && !tags && (error || help) ? (
-          <Prefix>
-            {help && !error && (
-              <div className="flex" ref={iconRef}>
-                <InformationCircle color="info" size="small" />
-              </div>
-            )}
-            {error && (
-              <div className="flex" ref={iconRef}>
-                <AlertCircle color="critical" size="small" />
-              </div>
-            )}
-          </Prefix>
-        ) : (
-          prefix && <Prefix>{prefix}</Prefix>
-        )}
-        {label && inlineLabel && (
-          <div
-            className={cx(
-              "flex items-center justify-center",
-              "pointer-events-none h-full",
-              "[&>.orbit-form-label]:mb-0",
-              "[&>.orbit-form-label]:text-form-element-normal [&>.orbit-form-label]:whitespace-nowrap [&>.orbit-form-label]:leading-normal",
-              "[&>.orbit-form-label]:z-[3]",
-              !tags && (error || help) ? "ps-100" : "ps-300",
-            )}
-            ref={labelRef}
-          >
-            <FormLabel required={required} error={!!error} help={!!help} inlineLabel={inlineLabel}>
-              {label}
-            </FormLabel>
-          </div>
-        )}
+        <label className="relative z-[2] flex items-center" ref={fieldRef} htmlFor={inputId}>
+          {inlineLabel && !tags && (error || help) ? (
+            <Prefix>
+              {help && !error && (
+                <span className="flex" ref={iconRef}>
+                  <InformationCircle color="info" size="small" />
+                </span>
+              )}
+              {error && (
+                <span className="flex" ref={iconRef}>
+                  <AlertCircle color="critical" size="small" />
+                </span>
+              )}
+            </Prefix>
+          ) : (
+            prefix && <Prefix>{prefix}</Prefix>
+          )}
+          {label && inlineLabel && (
+            <span
```

Comment:
I've also replaced some divs by spans, because of the "phrasing content" rule (label should not include divs, etc..)